src/server/handlers/launchpad.js

```diff
@@ -513,85 +490,85 @@ export const getSnapBuilds = (req, res) => {
-        return lpClient.patch(snap.self_link, { auto_build: true })
-          .then(() => builds);
-      } else {
-        return builds;
-      }
-    });
+  const builds = await lpClient.named_post(
+    snap.self_link, 'requestAutoBuilds'
+  );
+  if (!snap.auto_build) {
+    await lpClient.patch(snap.self_link, { auto_build: true });
+  }
+  return builds;
 };
 
-export const requestSnapBuilds = (req, res) => {
-  checkAdminPermissions(req.session, req.body.repository_url)
-    .then(() => internalFindSnap(req.body.repository_url))
-    .then(internalRequestSnapBuilds)
-    .then((builds) => {
-      return res.status(201).send({
-        status: 'success',
-        payload: {
-          code: 'snap-builds-requested',
-          builds: builds
-        }
-      });
-    })
-    .catch((error) => sendError(res, error));
+export const requestSnapBuilds = async (req, res) => {
+  try {
+    await checkAdminPermissions(req.session, req.body.repository_url);
+    const snap = await internalFindSnap(req.body.repository_url);
+    const builds = await internalRequestSnapBuilds(snap);
+    return res.status(201).send({
+      status: 'success',
+      payload: {
+        code: 'snap-builds-requested',
+        builds: builds
+      }
+    });
+  } catch (error) {
+    return sendError(res, error);
+  }
 };
 
-export const deleteSnap = (req, res) => {
-  let owner;
-  checkAdminPermissions(req.session, req.body.repository_url)
-    .then((result) => {
-      owner = result.owner;
-      return internalFindSnap(req.body.repository_url);
-    })
-    .then((snap) => snap.lp_delete())
-    .then(() => {
-      const urlPrefix = getRepoUrlPrefix(owner);
-      const prefixCacheId = getUrlPrefixCacheId(urlPrefix);
-      const repoCacheId = getRepositoryUrlCacheId(req.body.repository_url);
-
-      return getMemcached().del(prefixCacheId)
-        .catch((err) => {
-          logger.error(`Error deleting ${prefixCacheId} from memcached:`, err);
-        })
-        .then(() => getMemcached().del(repoCacheId))
-        .catch((err) => {
-          logger.error(`Error deleting ${repoCacheId} from memcached:`, err);
-        })
-        .then(() => {
-          return res.status(200).send({
-            status: 'success',
-            payload: {
-              code: 'snap-deleted',
-              message: 'Snap deleted'
-            }
-          });
-        });
-    })
-    .catch((error) => sendError(res, error));
+export const deleteSnap = async (req, res) => {
+  try {
+    const { owner } = await checkAdminPermissions(
+      req.session, req.body.repository_url
+    );
+    const snap = await internalFindSnap(req.body.repository_url);
+    await snap.lp_delete();
+
+    const urlPrefix = getRepoUrlPrefix(owner);
+    const prefixCacheId = getUrlPrefixCacheId(urlPrefix);
+    const repoCacheId = getRepositoryUrlCacheId(req.body.repository_url);
+    try {
+      await getMemcached().del(prefixCacheId);
+    } catch (error) {
```

Comment:
In a simple cases like these, can we actually keep the `.catch()` notation from promise (which should be allowed on await calls)?

```js
await getMemcached().del(prefixCacheId)
  .catch((error) => logger.error( ... ));
```

This will allow to avoid repeated and nested `try ... catch ...`.

I'm not sure if I've seen any other case apart from `getMemcached().del()` where it was so visible, but maybe there are some.